rapid7_insightvm_cloud/icon_rapid7_insightvm_cloud/util/api.py

```diff
@@ -0,0 +1,75 @@
+import json
+
+from requests import request
+from logging import Logger
+
+import insightconnect_plugin_runtime
+from insightconnect_plugin_runtime.exceptions import PluginException
+from requests.exceptions import HTTPError
+
+
+class IVM_Cloud:
+    def __init__(self, token: str, logger: Logger, url: str):
+        self.logger = logger
+        self.token = token
+        self.base_url = url
+
+    def call_api(self, path: str, request_type: str, params: dict = None, body: dict = None):
+        if params is None:
+            params = {}
+        if body is None:
+            body = {}
+
+        api_url = self.base_url + path
+
+        headers = {"x-api-key": self.token, "content-type": "application/json"}
+
+        try:
+            response = request(
+                request_type,
+                self.base_url + path,
+                params=insightconnect_plugin_runtime.helper.clean(params),
+                headers=headers,
+                data=json.dumps(body),
+            )
+            if response.status_code not in [200, 201, 202]:
+                raise PluginException(
+                    cause=f"Failed to get a valid response from InsightVM at endpoint {api_url}",
+                    assistance=f"Response was {response.request.body}",
+                    data=response.status_code,
+                )
+            if response.text == "":
```

Comment:
Returning different values based on the response makes it less clear to know what the return value from the function might be. It might be nice to have a standard type that includes the body (if there was one) and the status.  (or perhaps just return the requests response type). With type annotations, for the return value, it would be clear what the expected response is and how to use it.